modules/Bio/EnsEMBL/Hive/HivePipeline.pm

```diff
@@ -544,148 +564,183 @@ sub print_diagram {
                 } else {
-                    print "Tweak.Changing\tpipeline.$attrib_name ::\t$old_value --> $new_value_str\n";
+                    $tweakStructure->{Return}->{NewValue} = $new_value_str;
+                    push @response, "Tweak.Changing\tpipeline.$attrib_name ::\t$old_value --> $new_value_str\n";
 
                     $self->$attrib_name( $new_value_str );
                     $need_write = 1;
                 }
 
             } else {
-                print "Tweak.Error   \tCould not find the pipeline-wide '$attrib_name' method\n";
+                $tweakStructure->{Error} = $self->{TWEAK_ERROR_MSG}->{FIELD_ERROR};
+                push @response, "Tweak.Error   \tCould not find the pipeline-wide '$attrib_name' method\n";
             }
-
+            push @{$responseStructure->{Tweaks}}, $tweakStructure;
         } elsif($tweak=~/^analysis\[([^\]]+)\]\.param\[(\w+)\](\?|#|=(.+))$/) {
             my ($analyses_pattern, $param_name, $operator, $new_value_str) = ($1, $2, $3, $4);
-
             my $analyses = $self->collection_of( 'Analysis' )->find_all_by_pattern( $analyses_pattern );
-            print "Tweak.Found   \t".scalar(@$analyses)." analyses matching the pattern '$analyses_pattern'\n";
+            push @response, "Tweak.Found   \t".scalar(@$analyses)." analyses matching the pattern '$analyses_pattern'\n";
 
             my $new_value = destringify( $new_value_str );
             $new_value_str = stringify( $new_value );
 
             foreach my $analysis (@$analyses) {
                 my $analysis_name = $analysis->logic_name;
-
                 my $old_value = $analysis->parameters;
-
                 my $param_hash  = destringify( $old_value );
+                my $tweakStructure;
+                $tweakStructure->{Object}->{Type} = $self->{TWEAK_OBJECT_TYPE}->{ANALYSIS};
+                $tweakStructure->{Action} = $self->{TWEAK_ACTION}->{substr($operator, 0, 1)};
+                $tweakStructure->{Object}->{Id} = $analysis->dbID + 0;
+                $tweakStructure->{Object}->{Name} = $analysis_name;
+                $tweakStructure->{Return}->{Field} = $param_name;
+                $tweakStructure->{Return}->{OldValue} =  exists($param_hash->{ $param_name }) ? stringify($param_hash->{ $param_name }) : undef;
 
                 if($operator eq '?') {
-                    print "Tweak.Show    \tanalysis[$analysis_name].param[$param_name] ::\t"
-                        . (exists($param_hash->{ $param_name }) ? stringify($param_hash->{ $param_name }) : '(missing value)')
-                        ."\n";
+                    $tweakStructure->{Return}->{NewValue} = $tweakStructure->{Return}->{OldValue};
+                    push @response, "Tweak.Show    \tanalysis[$analysis_name].param[$param_name] ::\t"
+    	                . (exists($param_hash->{ $param_name }) ? stringify($param_hash->{ $param_name }) : '(missing value)')
```

Comment:
Still not quite right :(

Reply:
Sorry, something was wrong with my editor, it shows in another way tan it is in reality. Now checked in different editor and can see the difference.